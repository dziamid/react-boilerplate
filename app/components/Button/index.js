import React, { PropTypes } from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import IconButton from 'material-ui/IconButton';
import styles from './styles.css';
import classNames from 'classnames/bind';
const cx = classNames.bind(styles);

function Button({ className, icon, label, raised, children, onClick, ...rest }) {
  const isLabelledButton = label !== undefined || children !== undefined;
  const isIconOnlyButton = !isLabelledButton && icon;
  const ButtonComponent = raised ? RaisedButton : FlatButton;

  return (
    <div className={cx(styles.root, className)}>

      { isIconOnlyButton ?
        <IconButton onClick={onClick} {...rest}>
          { icon }
        </IconButton>
        : null
      }

      { isLabelledButton ?
        <ButtonComponent onClick={onClick} label={children || label} {...rest} />
        : null
      }

    </div>
  );
}

Button.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node,
  icon: PropTypes.object,
  label: PropTypes.string,
};

export default Button;
