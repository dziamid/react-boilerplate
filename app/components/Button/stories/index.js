import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import Button from '../index';

storiesOf('Button', module)
  .add('with text', () => (
    <Button onClick={action('clicked')}>Hello Button</Button>
  ))
  .add('with some emoji', () => (
    <Button onClick={action('clicked')}>😀 😎 👍 💯</Button>
  ));
