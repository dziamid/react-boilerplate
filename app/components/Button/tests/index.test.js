/**
 * Testing our Button component
 */

import Button from '../index';

import { shallow } from 'enzyme';
import React from 'react';

describe('<Button />', () => {
  it('should render its children', () => {
    const children = (<h1>Test</h1>);
    const renderedComponent = shallow(
      <Button href="http://mxstbr.com">
        {children}
      </Button>
    );
    expect(renderedComponent.contains(children)).toBe(true);
  });

  it('should adopt the className', () => {
    const renderedComponent = shallow(<Button className="test"><h1>Test</h1></Button>);
    expect(renderedComponent.find('a').hasClass('test')).toBe(true);
  });

  it('should render an <a> tag if no route is specified', () => {
    const renderedComponent = shallow(<Button href="http://mxstbr.com"><h1>Test</h1></Button>);
    expect(renderedComponent.find('a').length).toBe(1);
  });

  it('should render a button to change route if the handleRoute prop is specified', () => {
    const renderedComponent = shallow(<Button handleRoute={function handler() {}}><h1>Test</h1></Button>);

    expect(renderedComponent.find('button').length).toBe(1);
  });

  it('should handle click events', () => {
    const onClickSpy = jest.fn();
    const renderedComponent = shallow(<Button onClick={onClickSpy}><h1>Test</h1></Button>);
    renderedComponent.find('a').simulate('click');
    expect(onClickSpy).toHaveBeenCalled();
  });
});
