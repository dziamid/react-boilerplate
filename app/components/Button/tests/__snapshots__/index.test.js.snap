exports[`<Button /> should render Button with label 1`] = `
<FlatButton
  disabled={false}
  label="Button with label"
  labelPosition="after"
  labelStyle={Object {}}
  onClick={[Function]}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  primary={false}
  secondary={false} />
`;

exports[`<Button /> should render Button with svg icon 1`] = `
<IconButton
  disableTouchRipple={false}
  disabled={false}
  iconStyle={Object {}}
  onClick={[Function]}
  tooltipPosition="bottom-center"
  touch={false}>
  <NavigationClose />
</IconButton>
`;
